src/core/coap/coap.cpp

```diff
@@ -672,20 +692,32 @@ void Coap::ProcessReceivedRequest(Header &aHeader, Message &aMessage, const Ip6:
         if (strcmp(resource->mUriPath, uriPath) == 0)
         {
             resource->HandleRequest(aHeader, aMessage, aMessageInfo);
+            error = kThreadError_None;
             ExitNow();
         }
     }
 
     if (mDefaultHandler)
     {
         mDefaultHandler(mDefaultHandlerContext, &aHeader, &aMessage, &aMessageInfo);
+        error = kThreadError_None;
```

Comment:
Could you elaborate more on how you want to update the default handler? I am not sure I'm following.
Also can't this be solved by setting a "default" default handler, which just sends 404?
To give more context, one of the reasons why I though of default handler was that CoAP and related standards have a quite a bunch of different requirements/responses/options and corner cases, which are (functionally) irrelevant to OpenThread and additional code would only bloat the code size. However if an application developer is limited by the OpenThread CoAP, they can extend it using the default handler.